File: ehs_frontend/src/api/quiz.ts
import { api } from './client'

export interface QuizQuestion {
  question_number: number
  question_type: 'mcq' | 'descriptive' | 'scenario'
  question_text: string
  options: string[] | null   // MCQ only
  correct_answer: string     // MCQ: "A"/"B"/"C"/"D" | descriptive/scenario: model answer text
  explanation: string | null
}
export interface QuizResponse { questions: QuizQuestion[] }

export async function generateQuiz(params: {
  activity: string
  hazard_type: string
  severity_likelihood: string
  moc_ppe: string
  remarks: string
}): Promise<QuizResponse> {
  const { data } = await api.post('/api/v1/quiz/generate', params)
  return data
}

File: ehs_backend/app/schemas/quiz.py
from typing import Literal
from pydantic import BaseModel


class QuizGenerateRequest(BaseModel):
    activity: str
    hazard_type: str
    severity_likelihood: str
    moc_ppe: str
    remarks: str = ""


class QuestionOut(BaseModel):
    question_number: int
    question_type: Literal["mcq", "descriptive", "scenario"]
    question_text: str
    options: list[str] | None        # MCQ only: ["A. ...", "B. ...", "C. ...", "D. ..."]
    correct_answer: str              # MCQ: "A"/"B"/"C"/"D" | descriptive/scenario: model answer text
    explanation: str | None


class QuizGenerateResponse(BaseModel):
    questions: list[QuestionOut]
